test/intersections.jl

```diff
@@ -103,51 +108,184 @@
-    @test intersection(s1, s3) |> type == OverlappingSegments
-    @test s1 ∩ s3 == Segment(P3(0.5, 0.0, 0.0), P3(1.0, 0.0, 0.0))
-    @test intersection(s1, s4) |> type == MidTouchingSegments
-    @test s1 ∩ s4 ≈ P3(0.0, 0.0, 0.0)
-    @test intersection(s1, s5) |> type == MidTouchingSegments
-    @test s1 ∩ s5 ≈ P3(0.0, 0.0, 0.0)
-    @test intersection(s1, s6) |> type == CornerTouchingSegments
-    @test s1 ∩ s6 ≈ P3(0.0, 0.0, 0.0)
-    @test intersection(s1, s7) |> type == NoIntersection
-    @test isnothing(s1 ∩ s7)
-    @test intersection(s1, s8) |> type == NoIntersection
-    @test isnothing(s1 ∩ s8)
-    @test intersection(s1, s9) |> type == NoIntersection
-    @test isnothing(s1 ∩ s9)
-    @test intersection(s1, s10) |> type == NoIntersection
-    @test isnothing(s1 ∩ s10)
-    @test intersection(s1, s11) |> type == NoIntersection
-    @test isnothing(s1 ∩ s11)
-    @test intersection(s1, s12) |> type == CornerTouchingSegments
-    @test s1 ∩ s12 ≈ P3(1.0, 0.0, 0.0)
+    s₁ = Segment(P3(0.0, 0.0, 0.0), P3(1.0, 0.0, 0.0))
+    s₂ = Segment(P3(0.5, 1.0, 0.0), P3(0.5, -1.0, 0.0))
+    s₃ = Segment(P3(0.5, 0.0, 0.0), P3(1.5, 0.0, 0.0))
+    s₄ = Segment(P3(0.0, 1.0, 0.0), P3(0.0, -2.0, 0.0))
+    s₅ = Segment(P3(-1.0, 1.0, 0.0), P3(2.0, -2.0, 0.0))
+    s₆ = Segment(P3(0.0, 0.0, 0.0), P3(0.0, 1.0, 0.0))
+    s₇ = Segment(P3(-1.0, 1.0, 0.0), P3(-1.0, -1.0, 0.0))
+    s₈ = Segment(P3(-1.0, 1.0, 1.0), P3(-1.0, -1.0, 1.0))
+    s₉ = Segment(P3(0.5, 1.0, 1.0), P3(0.5, -1.0, 1.0))
+    s₁₀ = Segment(P3(0.0, 1.0, 0.0), P3(1.0, 1.0, 0.0))
+    s₁₁ = Segment(P3(1.5, 0.0, 0.0), P3(2.5, 0.0, 0.0))
+    s₁₂ = Segment(P3(1.0, 0.0, 0.0), P3(2.0, 0.0, 0.0))
+
+    @test intersection(s₁, s₂) |> type == CrossingSegments
+    @test s₁ ∩ s₂ ≈ P3(0.5, 0.0, 0.0)
+    @test intersection(s₁, s₃) |> type == OverlappingSegments
+    @test s₁ ∩ s₃ == Segment(P3(0.5, 0.0, 0.0), P3(1.0, 0.0, 0.0))
+    @test intersection(s₁, s₄) |> type == MidTouchingSegments
+    @test s₁ ∩ s₄ ≈ P3(0.0, 0.0, 0.0)
+    @test intersection(s₁, s₅) |> type == MidTouchingSegments
+    @test s₁ ∩ s₅ ≈ P3(0.0, 0.0, 0.0)
+    @test intersection(s₁, s₆) |> type == CornerTouchingSegments
+    @test s₁ ∩ s₆ ≈ P3(0.0, 0.0, 0.0)
+    @test intersection(s₁, s₇) |> type == NoIntersection
+    @test isnothing(s₁ ∩ s₇)
+    @test intersection(s₁, s₈) |> type == NoIntersection
+    @test isnothing(s₁ ∩ s₈)
+    @test intersection(s₁, s₉) |> type == NoIntersection
+    @test isnothing(s₁ ∩ s₉)
+    @test intersection(s₁, s₁₀) |> type == NoIntersection
+    @test isnothing(s₁ ∩ s₁₀)
+    @test intersection(s₁, s₁₁) |> type == NoIntersection
+    @test isnothing(s₁ ∩ s₁₁)
+    @test intersection(s₁, s₁₂) |> type == CornerTouchingSegments
+    @test s₁ ∩ s₁₂ ≈ P3(1.0, 0.0, 0.0)
 
     # type stability tests
-    s1 = Segment(P2(0,0), P2(1,0))
-    s2 = Segment(P2(0.5,0.0), P2(2,0))
-    @inferred someornone(s1, s2)
+    s₁ = Segment(P2(0,0), P2(1,0))
+    s₂ = Segment(P2(0.5,0.0), P2(2,0))
```

Comment:
It is usually a good idea, but in this case the code style of the surrounding code is already defined. More importantly, the code changes that are proposed here are very delicate and we would like to keep our history of commits clean so that people understand what happened in the past. Changes to these tests would just pollute the history for someone that is just trying to debug a problem.